main.go

```diff
@@ -174,6 +187,28 @@ func main() {
 }
 
 func startLeaderElection(leaderElectionClient kubernetes.Interface, recorder record.EventRecorder, run func(ctx context.Context)) {
+	gwLeadershipConfig := leaderConfig{}
+
+	err := envconfig.Process(leadershipConfigEnvPrefix, &gwLeadershipConfig)
+	if err != nil {
+		klog.Fatalf("error processing environment config for %s: %v", leadershipConfigEnvPrefix, err)
+	}
```

Comment:
I wonder if we should start using a ConfigMap for this type of config ...

Reply:
There are many params that can be moved to a ConfigMap. Is it ok if we take it up in a separate patch?

Reply:
I think it’s something we should do, and also the secrets. There’s an open issue for that. May be something we can do after this release

Reply:
+1